### PR TITLE
Upgrade metrics integration to incorporate changes from JAVA-2808 (fixes #117)

### LINKED ISSUE #117: Upgrade metrics integration to incorporate changes from JAVA-2808
JAVA-2808 introduced metrics bindings for MicroProfile into the driver. Upgrade the Quarkus extension to use those bindings instead of the custom logic introduced earlier.

### PR DESCRIPTION
**Pull request by tomekl007**

### CODE REVIEW

File: deployment/src/test/java/com/datastax/oss/quarkus/deployment/internal/metrics/CassandraMetricsTest.java [hunk 1]
```diff
@@ -15,7 +15,8 @@
  */
 package com.datastax.oss.quarkus.deployment.internal.metrics;
 
-import static com.datastax.oss.quarkus.runtime.internal.metrics.MicroProfileMetricsUpdater.CASSANDRA_METRICS_PREFIX;
+import static com.datastax.oss.quarkus.runtime.internal.context.QuarkusDriverContext.*;
+import static com.datastax.oss.quarkus.runtime.internal.session.DefaultQuarkusCqlSession.*;
```
Comment (absurdfarce) [inline deployment/src/test/java/com/datastax/oss/quarkus/deployment/internal/metrics/CassandraMetricsTest.java:19]: This import looks to be unused

File: deployment/src/test/java/com/datastax/oss/quarkus/deployment/internal/metrics/CassandraMetricsTest.java [hunk 2]
```diff
@@ -15,7 +15,8 @@
  */
 package com.datastax.oss.quarkus.deployment.internal.metrics;
 
-import static com.datastax.oss.quarkus.runtime.internal.metrics.MicroProfileMetricsUpdater.CASSANDRA_METRICS_PREFIX;
+import static com.datastax.oss.quarkus.runtime.internal.context.QuarkusDriverContext.*;
```
Comment (absurdfarce) [inline deployment/src/test/java/com/datastax/oss/quarkus/deployment/internal/metrics/CassandraMetricsTest.java:18]: Maybe avoid the star import with just:

```java
import static com.datastax.oss.quarkus.runtime.internal.context.QuarkusDriverContext.CASSANDRA_METRICS_PREFIX;
```

File: runtime/src/main/java/com/datastax/oss/quarkus/runtime/internal/context/QuarkusDriverContext.java [hunk 3]
```diff
@@ -55,4 +62,10 @@ protected NettyOptions buildNettyOptions() {
       return super.buildNettyOptions();
     }
   }
+
+  @NonNull
+  @Override
+  public String getSessionName() {
+    return CASSANDRA_METRICS_PREFIX + "." + super.getSessionName();
+  }
```
Comment (absurdfarce) [inline runtime/src/main/java/com/datastax/oss/quarkus/runtime/internal/context/QuarkusDriverContext.java:70]: Is the goal of this change to distinguish java-driver sessions from Quarkus sessions?  Or are you looking to accomplish something else here?

**Comment by absurdfarce** [comment]
Mostly looks good, a few nits and a question but this looks pretty good

File: runtime/src/main/java/com/datastax/oss/quarkus/runtime/internal/context/QuarkusDriverContext.java (hunk 3 above)
Comment (tomekl007) [reply runtime/src/main/java/com/datastax/oss/quarkus/runtime/internal/context/QuarkusDriverContext.java:70]: yes, we want all driver metrics to be under the specific prefix: `cassandra` in this case

**Pull request merged**